core/agent/htask.go

```diff
@@ -339,20 +340,37 @@ func listenActionsAsync(brpopCtx context.Context, complete chan int) {
 		OnConnect: setClientNameCallback,
 	})
 
+	// Ignore the credential error on agent startup
+	//
+	// Initialize to a well-known error condition, to avoid log pollution
+	// and false alarms. When an agent is created its credentials might be
+	// still not stored in the leader node Redis instance:
+	var lastPopErr error = errors.New("WRONGPASS invalid username-password pair or user is disabled.")
+
 	for { // Action listen loop
 		var task models.Task
 
 		// Pop the task from the agent tasks queue
 		popResult, popErr := rdb.BRPop(brpopCtx, pollingDuration, agentPrefix+"/tasks").Result()
 		if popErr == redis.Nil {
+			// poll timeout, it's ok: start a new cycle
+			lastPopErr = nil // Clear the last error buffer
 			continue
 		} else if brpopCtx.Err() != nil {
 			break
 		} else if popErr != nil {
-			log.Print(SD_ERR+"Task queue pop error: ", popErr)
+			// Avoid error log repetitions: print the error only if it is
+			// different from the value in the last error buffer
+			if lastPopErr == nil || popErr.Error() != lastPopErr.Error() {
```

Comment:
Why are you checking if `lastPopErr` is null? I do not know all the code, but it seems to me is not useful: if the current error is different from the last one, just print it
```suggestion
			if popErr.Error() != lastPopErr.Error() {
```